apps/spruce/src/components/VersionRestartModal/VersionRestartModal.tsx

```diff
@@ -209,6 +219,29 @@ const getTaskIds = (selectedTasks: versionSelectedTasks) =>
     }))
     .filter(({ taskIds }) => taskIds.length > 0);
 
+const getNumEstimatedActivatedGeneratedTasks = (
```

Comment:
I see this function declared 3 times in this pr with slightly different implementations but they all seem to be doing the same thing. Can you make it more generic and reuse it as a util method instead of rewriting it multiple times?